pkg/secrets/scrub_test.go

```diff
@@ -36,6 +36,160 @@ func TestScrubFromNestedStructPointer(t *testing.T) {
 	assert.Equal(t, "name", testStruct.Name)
 }
 
+func TestScrubMapFromStruct(t *testing.T) {
+	testStruct := &toplevel{
+		Map:      map[string]string{"client_secret": "scrub", "keep": "keep_top"},
+		ScrubMap: map[string]string{"client_secret": "scrub", "keep": "keep_top"},
+		ConfigPtr: &config{
+			Map:      map[string]string{"client_secret": "scrub", "keep": "keep_ptr"},
+			ScrubMap: map[string]string{"client_secret": "scrub", "keep": "keep_ptr"},
+		},
+		Config: config{
+			Map:      map[string]string{"client_secret": "scrub", "keep": "keep"},
+			ScrubMap: map[string]string{"client_secret": "scrub", "keep": "keep"},
+		},
+	}
+	ScrubSecretsFromStructWithReplacement(testStruct, "***")
+
+	assert.Equal(t, map[string]string{"client_secret": "scrub", "keep": "keep_top"}, testStruct.Map)
+	assert.Equal(t, map[string]string{"client_secret": "***", "keep": "keep_top"}, testStruct.ScrubMap)
+	assert.Equal(t, map[string]string{"client_secret": "scrub", "keep": "keep_ptr"}, testStruct.ConfigPtr.Map)
+	assert.Equal(t, map[string]string{"client_secret": "***", "keep": "keep_ptr"}, testStruct.ConfigPtr.ScrubMap)
+	assert.Equal(t, map[string]string{"client_secret": "scrub", "keep": "keep"}, testStruct.Config.Map)
+	assert.Equal(t, map[string]string{"client_secret": "***", "keep": "keep"}, testStruct.Config.ScrubMap)
+}
+
+func TestScrubMapFromStructWithoutChange(t *testing.T) {
```

Comment:
Perhaps we can go for full coverage of "empty map" case and have two test cases here:

```
testStruct := &toplevel{
		Map:      map[string]string{"keep": "keep_top"},
		ScrubMap: map[string]string{"keep": "keep_top"},
		ConfigPtr: &config{
			Map:      map[string]string{"keep": "keep_ptr"},
			ScrubMap: map[string]string{"keep": "keep_ptr"},
		},
		Config: config{
			Map:      map[string]string{"keep": "keep_ptr"},
			ScrubMap: map[string]string{"keep": "keep_ptr"},
		},
	}
```

and 

```
testStruct := &toplevel{
		Map:      map[string]string{},
		ScrubMap: map[string]string{},
		ConfigPtr: &config{
			Map:      map[string]string{},
			ScrubMap: map[string]string{},
		},
		Config: config{
			Map:      map[string]string{},
			ScrubMap: map[string]string{},
		},
	}
```